packages/server/modules/gatekeeper/services/subscriptions.ts

```diff
@@ -416,23 +431,48 @@ export const upgradeWorkspaceSubscriptionFactory =
     if (!workspaceSubscription) throw new WorkspaceSubscriptionNotFoundError()
 
     const planOrder: Record<PaidWorkspacePlans, number> = {
+      // old
       business: 3,
       plus: 2,
-      starter: 1
+      starter: 1,
+      // new
+      team: 1,
+      pro: 2
+    }
+
+    if (
+      !FF_WORKSPACES_NEW_PLANS_ENABLED &&
+      (isNewPlanType(workspacePlan.name) || isNewPlanType(targetPlan))
+    ) {
+      throw new NotImplementedError()
+    }
+
+    const planCheckers = [isNewPlanType, isOldPlanType]
+    for (const isSpecificPlanType of planCheckers) {
+      const oldPlanFitsSchema = isSpecificPlanType(workspacePlan.name)
+      const newPlanFitsSchema = isSpecificPlanType(targetPlan)
+      if (xor(oldPlanFitsSchema, newPlanFitsSchema)) {
+        throw new WorkspacePlanUpgradeError(
+          'Attempting to switch between incompatible plan types'
+        )
+      }
     }
 
     if (
```

Comment:
throw here for any of the new plan type, we're not supporting upgrades yet. Need to figure out the productId priceId calcualtion logic